Fixes for dependency breaking changes (flutter_bloc and flutter_facebook_login)

cloudwheels opened the pull request:
(Temporarily) fixes breaking changes in new versions plugins:
1. flutter_bloc
By pinning version to 0.16.0
2. fluttter_facebook_login
By updating call to logInWithReadPermission() with logIn() in login_bloc.dart

Issue created at https://github.com/nstosic/toptal-flutter-chat/issues/5

cloudwheels committed:
peg flutter_bloc to 0.22.0 to prevent v1 breaking change - dispatch() deprecated

cloudwheels committed:
rename facebookSignInRepo logInWithReadPermission to logIn() breaking change flutter_facebook_login 3.0.0

nstosic commented:
Thanks for the PR @cloudwheels . However I'd rather we keep the repo with the latest libraries than pinning the dependency to a particular version, in accordance with Google's Abseil's "Live at the head" philosophy. Although the dependency itself should provide tools to migrate to their latest version, we only have control over how we decide to move on. I'll see if I can update the repo to latest `flutter_bloc` and `flutter_facebook_login`. If it turns out to be too much work, I'll approve your PR. Otherwise, I'll open another with the library update.

cloudwheels committed:
Update distributionUrl in gradle-wrapper.properties

Update distributionUrl in gradle-wrapper.properties to 5.5.1 to prevent errors with shared_preferences plugin dependency

cloudwheels commented:
No problem and thanks for this repo - a very helpful starter! Yes, this is intended as a temporary fix and it's unfortunate that the fluter_bloc plugin introduces these breaking changes. I have just pushed another commit to this branch to update the gradle version which I had to do also (sometimes it's hard to tell where the errors are actually arising once you start changing plugin versions!)

nstosic commented:
Hi @cloudwheels I've opened #7 .

Try checking out my branch `bugfix/flutter_bloc_breaking_change` and let me know if it works for you. Afterwards I'll merge it.

cloudwheels commented:
@nstosic away for the weekend but will do! I've also implemented a trigger
function for notifications, so will share that too...

On Sat, 26 Oct 2019, 09:16 Nemanja Stošić, <notifications@github.com> wrote:

> Hi @cloudwheels <https://github.com/cloudwheels> I've opened #7
> <https://github.com/nstosic/toptal-flutter-chat/pull/7> .
>
> Try checking out my branch bugfix/flutter_bloc_breaking_change and let me
> know if it works for you. Afterwards I'll merge it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/nstosic/toptal-flutter-chat/pull/6?email_source=notifications&email_token=AKTEKOGMDLAQKCAGZP6ZK6LQQP4GFA5CNFSM4JERONG2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOECKCY4Y#issuecomment-546581619>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKTEKOELES75F5OPKBLCKWTQQP4GFANCNFSM4JERONGQ>
> .
>
